docs/content/docs/start/react-tutorial.md

```diff
@@ -0,0 +1,295 @@
+        }
+    }, [fluidSharedMap])
+    ```
+
+1. Replace `TODO 4` with the following code. Note about this code:
+
+    - The Fluid `SharedObject.get` method returns the data of the `SharedObject` (in this case the `SharedMap` object), which is roughly the `SharedObject` without any of its methods. So the `setLocalTimestamp` function is setting the `localTimestamp` state to a copy of the data of the `SharedMap` object. (The key "time" that is passed to `SharedObject.get` is created in a later step. It will have been set by the time this code runs the first time.)
+    - `updateLocalTimestamp` is called immediately to ensure that `localTimestamp` is initialized with the current shared timestamp value.
+
+    ```js
+    const { sharedTimestamp } = fluidSharedMap;
+    const updateLocalTimestamp = () => setLocalTimestamp({ time: sharedTimestamp.get("time") });
+    
+    updateLocalTimestamp();
+    ```
+
+1. To ensure that the `localTimestamp` state is updated whenever the `fluidSharedMap` is changed _even by other clients_, replace `TODO 5` with the following code. Note that because `updateLocalTimestamp` calls the state-setting function `setTimestamp`, a rerender is triggered whenever any client changes the Fluid `fluidSharedMap`.
+
+    ```js
+    sharedTimestamp.on("valueChanged", updateLocalTimestamp);
+   ```
+
+1. It is a good practice to deregister event handlers when the React component dismounts, so replace `TODO 6` with the following code.
+
+    ```js
+    return () => { sharedTimestamp.off("valueChanged", updateLocalTimestamp) }
+   ```
+
+### Create the view
+
+Below the `useEffect` hook, replace the `return ();` line with the following code. Note about this code:
+
+- If the `localTimestamp` state has not been initialized, a blank screen is rendered.
+- The `sharedTimestamp.set` method sets the _key_ of the `fluidSharedMap` object to "time" and the _value_ to the current UNIX epoch time. This triggers the `valueChanged` event on the object, so the `updateLocalTimestamp` function runs and sets the `localTimestamp` state to the same object; for example, `{time: "1615996266675"}`. The `App` component rerenders and the `<span>` is updated with the latest timestamp.
+- All other clients update too because the Fluid server propagates the change to the `fluidSharedMap` on all of them and this `valueChanged` event updates the `localTimestamp` state on all of them.
+
+```js
+if (localTimestamp) {
+    return (
+        <div className="App">
+            <button onClick={() => fluidSharedMap.sharedTimestamp.set("time", Date.now().toString())}>
+                Get Time
+            </button>
+            <span>{localTimestamp.time}</span>
+        </div>
+    )
+} else {
+     return <div/>;
+}
+```
+
+## Start the Fluid server and run the application
+
+In the Command Prompt, run the following command to start the FRS server. Note that `tinylicious` is the name of the FRS that runs on localhost.
```

Comment:
This isn't quite correct with respect to Tinylicious. Tinylicious and FRS are separate, but the `FrsClient` can talk to either Tinylicious or FRS. When testing locally, you can configure FrsClient to talk to Tinylicious, and you run Tinylicious separately as described here. [This info](https://github.com/microsoft/FluidHelloWorld#accessing-fluid-data) might be helpful too. I'm working on getting that stuff into these docs.

Reply:
Hmmm. The info you linked to says "FrsClient supports deployed Azure Fluid Relay service instances for production purposes, as well as a local, in-memory service instance, called Tinylicious, for development purposes." It seems to me that the "local in-memory service" should be called **a** Fluid Relay service. It's just confusing to have a object called FrsClient that sometimes connects to a thing called FRS and sometimes connects to a different thing that's NOT called FRS (Tinylicious).  If we don't want to call Tinylicious **an** FRS, then it would be better to go back to "TinyliciousClient" instead of "FrsClient".

Reply:
> It seems to me that the "local in-memory service" should be called **a** Fluid Relay service.

You should think of FRS as a brand name. The generic thing is a "Fluid service." Tinylicious is a Fluid service, in the same sense that FRS and Routerlicious are Fluid services.

We could indeed force the developer to use the TinyliciousClient whenever they're doing local development (which will always be against Tinylicious). That was the original plan, but we moved away from it because it's an unnecessary step. We can today trivially support both Tinylicious locally and FRS remotely in the same client though this may not be true forever if FRS and Tinylicious diverge (no plans there to be clear).

Fast-forwarding to when there are other non-Microsoft Fluid services, then their clients may behave differently, and it's not required that a service client support both local and remote development. In fact, another Fluid service may choose not to use Tinylicious at all, and instead provide their developers with another solution for local development (a docker image of their production service for example).

Reply:
@tylerbutler OK. I'll adjust the wording accordingly, so Tinylicious is described as a Fluid service, not a Fluid Relay service. But since FRS is a brand name, and the generic term is "Fluid service", I strongly recommend that we change the type FrsClient to FsClient. It is confusing to have a type whose name implies that it works only with the brand FRS when it actually can work with a non-FRS Fluid service.

Reply:
> I strongly recommend that we change the type FrsClient to FsClient. It is confusing to have a type whose name implies that it works only with the brand FRS when it actually can work with a non-FRS Fluid service.

We are planning to change the name, but it will still be the Azure-specific client. Would this be less confusing if the local server was called Local Azure Fluid Relay service instead of Tinylicious?

Reply:
> We are planning to change the name, but it will still be the Azure-specific client. Would this be less confusing if the local server was called Local Azure Fluid Relay service instead of Tinylicious?

Yes! "Tinylicious" and "Routerlicious" are not self-documenting. (But maybe just "Local Fluid Relay service" because it's not really Azure if it's local.

Reply:
This gets tricky, because there isn't just _one_ local Fluid service, and there will possibly be more in the future depending on how other folks set up their Fluid service offerings. The services we have differ on a variety of axes, so we don't typically refer to things generically, "this is the local test server" because that's not precise enough on our side. That's why we call the servers by proper names.

But this need not bleed out into the general developer story. We could either rename tinylicious, export it in another package (e.g. @fluidframework/local-service), or figure something else out. @skylerjokiel @danielroney FYI - any thoughts on this?